cli/api/commands/install.ts

```diff
@@ -1,11 +1,55 @@
+import * as fs from "fs";
+import { load as loadYaml, YAMLException } from "js-yaml";
 import * as path from "path";
 import { promisify } from "util";
 
 import * as childProcess from "child_process";
+import { dataform } from "df/protos/ts";
 
-export async function install(projectDir: string, skipInstall?: boolean) {
+export async function install(projectPath: string, skipInstall?: boolean) {
   if (skipInstall) {
     return;
   }
-  await promisify(childProcess.exec)("npm i --ignore-scripts", { cwd: path.resolve(projectDir) });
+  const resolvedProjectPath = path.resolve(projectPath);
+  const workflowSettingsPath = path.join(resolvedProjectPath, "workflow_settings.yaml");
+  const packageJsonPath = path.join(resolvedProjectPath, "package.json");
+  const packageLockJsonPath = path.join(resolvedProjectPath, "package-lock.json");
+
+  let installCommand = "npm i --ignore-scripts";
+
+  // Core's readWorkflowSettings method cannot be used for this because Core assumes that
+  // `require` can read YAML files directly.
+  const dataformCoreVersion = readDataformCoreVersionIfPresent(workflowSettingsPath);
+
+  if (dataformCoreVersion) {
+    // If there are other packages already in the package.json, specifying a specific package to
+    // install will trigger the other packages to be installed too.
```

Comment:
are you sure about this? it seems very surprising to me, and it does not seem to be in the docs: https://docs.npmjs.com/cli/v10/commands/npm-install

Reply:
I was surprised too, but it happened when I tried it!